shop/permissions.py

```diff
@@ -28,3 +28,18 @@ def wrapper(self, *method_args, **method_kwargs):
             return http_method(self, *method_args, **method_kwargs)
         return wrapper
     return check_permissions_decorator
+
+
+def add_and_check_permissions(permission_class):
```

Comment:
name of the function must correspond to the purpose of the function included in one verb